pkg/sql/planner.go

```diff
@@ -1015,7 +1015,8 @@ func (p *planner) StartLogicalReplicationJob(
 	// TODO(ssd): Name resolution needs to be thought through for
 	// the final syntax.
 	fullyQualifiedTableNames := make([]string, 0, len(tableNames))
-	for _, t := range tableNames {
+	repPairs := make([]jobspb.LogicalReplicationDetails_ReplicationPair, len(tableNames))
+	for i, t := range tableNames {
```

Comment:
Probably for the sql syntax discussion: i don't understand how we can figure out what tables the user wants us to replicate without them providing the fully resolved table names. Perhaps we assume that the user starts the replication job wile connected to the target database and provides a connection string to the accompanying source database. If that's the assumption we're using, it would be nice write that out here.

Reply:
@msbutler As of now, the 2 base syntax ideas that we have both assume the user starts the query on a database and provides the connection string to the other. Tentatively there is a option that I've noted down for explicit source and target db and table(maybe not needed unless we allow for differing table names). 

Using DT's syntax idea here as an example: `CREATE TABLE bar FROM REPLICATION FROM 'external://otherdb' WITH target_database=foo, source_database=blah`

Nothing final here, but this could then be resolved into blah.bar and foo.bar or at least is the idea.